[API Compatibility No.378] Add parameter alias support for tan

yyxxddjj opened the pull request:
### PR Category
Operator Mechanism

### PR Types
New features

### Description
PIR: migrate tan operators to new IR.
Add parameter alias support for tan.

zhwesky2010 commented on python/paddle/tensor/math.py:
```diff
@@ -115,12 +115,90 @@
 )
 
+
+def square(x=None, name=None, input=None, out=None):
```
这两个是在ops.py里实现，这里不要又导出来。而且这两个可以C++下沉。

yyxxddjj replied:
噢！我明白了

yyxxddjj commented:
@zhwesky2010 我已经更新了代码，我参考的是#76421中asin的实现方法，您有时间的时候，麻烦帮忙跑一下自动化检查，谢谢！

luotao1 commented:
请解决下冲突

yyxxddjj commented:
可以批准一下ci吗，

> > 感觉可以开始新一轮自动化检查了。
> 
> 可以考虑先做一个简单的任务，仓库中合入第一个PR后，该仓库后续的commit就会自动触发CI了。可以认领下面任务中的一个子任务。
> 
> * [📝 文档示例代码修复任务 #76545](https://github.com/PaddlePaddle/Paddle/issues/76545)

这个那边也在跑ci，所以我现在还不能自动触发ci，麻烦您了

yyxxddjj committed:
[API Compatibility No.378] Add parameter alias support for tan

Sink paddle.tan to cpp using default mapping to enable:
- input alias for x parameter
- out parameter support

Following the same pattern as asin PR #76421

yyxxddjj commented:
CI总是通过不了，我先把square的部分回退了，只保留了tan的修改，修改是完全仿照#76421中asin的下沉方法

luotao1 commented:
如果对[Paddle API兼容性增强](https://github.com/PaddlePaddle/community/blob/master/hackathon/hackathon_10th/%E3%80%90Hackathon_10th%E3%80%91%E9%A3%9E%E6%A1%A8%E6%8A%A4%E8%88%AA%E8%AE%A1%E5%88%92%E9%9B%86%E8%AE%AD%E8%90%A5%E9%A1%B9%E7%9B%AE%E5%90%88%E9%9B%86_%E6%8F%90%E5%89%8D%E6%89%B9.md#%E9%A1%B9%E7%9B%AE%E4%BA%8CPaddleAPI%E5%85%BC%E5%AE%B9%E6%80%A7%E5%A2%9E%E5%BC%BA)课题感兴趣，可以报名参加： @yyxxddjj 
- https://github.com/PaddlePaddle/Paddle/issues/76977
